code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm

```diff
@@ -6,113 +6,163 @@
+// *********** Essence Link
+// ***************************************
+/datum/action/xeno_action/activable/essence_link
+	name = "Essence Link"
+	action_icon_state = "healing_infusion"
+	mechanics_text = "Link to a xenomorph. This changes some of your abilities, and grants them and you both various bonuses."
+	cooldown_timer = 5 SECONDS
+	plasma_cost = 0
+	target_flags = XABB_MOB_TARGET
+	keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK
+	alternate_keybind_signal = COMSIG_XENOABILITY_ESSENCE_LINK_REMOVE
+	/// Used to determine whether there is an existing Essence Link or not. Also allows access to its vars.
+	var/datum/status_effect/stacking/essence_link/existing_link
+	/// The target of an existing link, if applicable.
+	var/mob/living/carbon/xenomorph/linked_target
+	/// Time it takes for the attunement levels to increase.
+	var/attunement_cooldown = 20 SECONDS
+
+/datum/action/xeno_action/activable/essence_link/can_use_ability(mob/living/carbon/xenomorph/target, silent = FALSE, override_flags)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!isxeno(target) || target.get_xeno_hivenumber() != X.get_xeno_hivenumber())
+		return FALSE
+	if(!X.Adjacent(target))
+		X.balloon_alert(X, "Not adjacent")
+		return FALSE
+	if(!existing_link && HAS_TRAIT(target, TRAIT_ESSENCE_LINKED))
+		target.balloon_alert(X, "She is already linked")
+		return FALSE
+	if(existing_link)
+		target.balloon_alert(X, "We are already linked")
+		return FALSE
+	return ..()
+
+/datum/action/xeno_action/activable/essence_link/use_ability(atom/target)
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!existing_link)
+		target.balloon_alert(X, "Linking...")
+		if(!do_after(X, DRONE_ESSENCE_LINK_WINDUP, TRUE, target, BUSY_ICON_FRIENDLY, BUSY_ICON_FRIENDLY))
+			X.balloon_alert(X, "Link cancelled")
+			return
+		X.apply_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK, 1, target)
+		existing_link = X.has_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+		linked_target = target
+		target.balloon_alert(target, "Essence Link established")
+	succeed_activate()
+
+/datum/action/xeno_action/activable/essence_link/alternate_action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	if(!existing_link)
+		X.balloon_alert(X, "No link to cancel")
+		return
+	end_ability()
+	return COMSIG_KB_ACTIVATED
+
+/// Ends the ability, removing signals and buffs.
+/datum/action/xeno_action/activable/essence_link/proc/end_ability()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/action/xeno_action/enhancement/enhancement_action = X.actions_by_path[/datum/action/xeno_action/enhancement]
+	enhancement_action.end_ability()
+	X.remove_status_effect(STATUS_EFFECT_XENO_ESSENCE_LINK)
+	existing_link = null
+	linked_target = null
+	add_cooldown()
+
+/datum/action/xeno_action/activable/essence_link/update_button_icon()
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/actions.dmi', button, "essence_link_[existing_link ? (existing_link.stacks) : (0)]")
+	return ..()
```

Comment:
This will need to be updated when MLGTASTICs pr is merged

Reply:
Understood.